upup/pkg/fi/secrets/clientset_secretstore.go

```diff
@@ -51,6 +52,44 @@ func NewClientsetSecretStore(clientset kopsinternalversion.KopsInterface, namesp
 	return c
 }
 
+func (c *ClientsetSecretStore) MirrorTo(basedir vfs.Path) error {
+	list, err := c.clientset.Keysets(c.namespace).List(v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("error listing keysets: %v", err)
+	}
+
+	for i := range list.Items {
+		keyset := &list.Items[i]
+
+		if keyset.Spec.Type != kops.SecretTypeSecret {
+			continue
+		}
+
+		primary := fi.FindPrimary(keyset)
+		if primary == nil {
+			glog.Warningf("skipping secret with no primary data: %s", keyset.Name)
```

Comment:
How would we have no primary data?

Reply:
It would be unexpected

Reply:
Then go boom please ... fmt.Error or panic